test/functional/p2p_addr_relay.py

```diff
@@ -207,6 +222,110 @@ def blocksonly_mode_tests(self):
 
         self.nodes[0].disconnect_p2ps()
 
+    def rate_limit_tests(self):
+
+        for contype, tokens, no_relay in [("outbound-full-relay", 1001, False), ("block-relay-only", 0, True), ("inbound", 1, False)]:
+            self.log.info(f'Test rate limiting of addr processing for {contype} peers')
+            self.stop_node(0)
+            os.remove(os.path.join(self.nodes[0].datadir, "regtest", "peers.dat"))
+            self.start_node(0, [])
+            self.mocktime = int(time.time())
+            self.nodes[0].setmocktime(self.mocktime)
+            if contype == "inbound":
+                peer = self.nodes[0].add_p2p_connection(AddrReceiver())
+            else:
+                peer = self.nodes[0].add_outbound_p2p_connection(AddrReceiver(), p2p_idx=0, connection_type=contype)
+
+            # Check that we start off with empty addrman
+            addr_count_0 = len(self.nodes[0].getnodeaddresses(0))
+            assert_equal(addr_count_0, 0)
+
+            # Send 600 addresses. For all but the block-relay-only peer this should result in at least 1 address.
+            addrs = 600
+            peer.send_and_ping(self.setup_rand_addr_msg(addrs))
+            peerinfo_1 = self.nodes[0].getpeerinfo()[0]
+            addrs_processed_1 = peerinfo_1['addr_processed']
+            addrs_rate_limited_1 = peerinfo_1['addr_rate_limited']
+            self.log.info(f"addrs_processed = {addrs_processed_1}, addrs_rate_limited = {addrs_rate_limited_1}")
+            if not no_relay:
+                assert_equal(addrs_processed_1, min(addrs, tokens))
+                assert_equal(addrs_rate_limited_1, addrs - min(addrs, tokens))
+            else:
+                assert_equal(addrs_processed_1, 0)
+                assert_equal(addrs_rate_limited_1, 0)
```

Comment:
This and similar statements below can be shortened:
```suggestion
            assert_equal(addrs_processed_1, min(addrs, tokens))
            assert_equal(addrs_rate_limited_1, 0 if no_relay else addrs - addrs_processed_1)
```